File: AssetSync/src/AssetSync.Core/Models/ConnectionStatus.cs
using AssetSync.Core.Enums;

namespace AssetSync.Core.Models;

public class ConnectionStatus
{
    public SourceSystem Service { get; set; }
    public ConnectionState State { get; set; }
    public string? Message { get; set; }
    public long? ResponseTimeMs { get; set; }
}

File: AssetSync-macOS/AssetSyncCore/Sources/AssetSyncCore/Models/ConnectionStatus.swift
import Foundation

public struct ConnectionStatus: Sendable {
    public var service: SourceSystem
    public var state: ConnectionState
    public var message: String?
    public var responseTimeMs: Int64?

    public init(service: SourceSystem, state: ConnectionState, message: String? = nil, responseTimeMs: Int64? = nil) {
        self.service = service
        self.state = state
        self.message = message
        self.responseTimeMs = responseTimeMs
    }
}
